src/services/music-providers/youtube-music/youtube-music.service.ts

```diff
@@ -0,0 +1,286 @@
+import {
+    add_playlist_items,
+    get_option,
+    get_playlist,
+    remove_playlist_items,
+    search,
+    setup,
+} from 'libmuse';
+import type { SearchSong } from 'libmuse/types/parsers/search.js';
+import type {
+    Playlist as PlaylistLibmuse,
+    EditPlaylistResult,
+    GetPlaylistOptions,
+} from 'libmuse/types/mixins/playlist.js';
+import type { SongArtist } from 'libmuse/types/parsers/songs.js';
+import type {
+    SearchOptions,
+    SearchResults,
+} from 'libmuse/types/mixins/search.js';
+
+import { Playlist, Store, Track } from '../../../entities.js';
+import { BaseMusicService } from '../base-music.service.js';
+import { LogService } from '../../log.service.js';
+import { LocalDbService } from '../../local-db.service.js';
+import { YoutubeMusicStore } from './youtube-music.store.js';
+import { checkIfArraysAreEqual } from '../../../utils/array.js';
+import { retry } from '../../../utils/retry.js';
+import { ConfigService } from '../../config.service.js';
+import { IConfig } from '../../../config/config.js';
+
+export class YoutubeMusicService extends BaseMusicService {
+    isReady = false;
+    private getPlaylistTracksLimit = 99999999999;
```

Comment:
The current `getPlaylistTracksLimit` value is set to an arbitrary large number, which is not a recommended practice. Instead, use JavaScript's `Number.MAX_VALUE` to represent the largest permissible number. This makes the code more readable, maintainable, and prevents possible overflow errors.